clash-lib/src/Clash/Backend/SystemVerilog.hs

```diff
@@ -1228,7 +1228,9 @@ expr_ _ (BlackBoxE pNm _ _ _ _ bbCtx _)
 expr_ _ (BlackBoxE pNm _ _ _ _ bbCtx _)
   | pNm == "Clash.Sized.Internal.Index.fromInteger#"
   , [Literal _ (NumLit n), Literal _ i] <- extractLiterals bbCtx
-  = exprLitSV (Just (Index (fromInteger n),fromInteger n)) i
+  , Just k <- clogBase 2 n
+  , let k' = max 1 k
```

Comment:
When would `k` be zero in this code path?

I feel like that only happens for types like `Index 1` which shouldn't render in the first place?

Reply:
I was basically copying what the VHDL backend does:
https://github.com/clash-lang/clash-compiler/blob/3b755b900810c727833875b1a15222d69d37029e/clash-lib/src/Clash/Backend/VHDL.hs#L1767-L1772

Reply:
But I think you're right

Reply:
That code was added in version 0.6.15 https://github.com/clash-lang/clash-compiler/commit/6b365dc082e78ed8c0f54f77faf30f62ae260a4a#diff-d636015ead3ec91b11144d7f18c90be9133810ab83457b3649a91e046cfae73fR647

Which might be from a time before we elided 0-bit values?

Reply:
Yeah probably.

I had seen that, at least in simple cases clash, removes `Index 1` literals because they're 0 bits wide values.
But I'm wasn't sure if there was some code path I didn't foresee. And assumed the author of that code in VHDL.hs knew what they were doing.
 
You want me to take one the `clogBase`?